rascal-vscode-extension/src/dap/RascalDebugView.ts

```diff
@@ -0,0 +1,116 @@
+        context.subscriptions.push(vscode.window.onDidOpenTerminal(fireEmitter));
+        context.subscriptions.push(vscode.window.onDidCloseTerminal(fireEmitter));
+        context.subscriptions.push(vscode.window.onDidChangeActiveTerminal(fireEmitter));
+        context.subscriptions.push(vscode.debug.onDidStartDebugSession(fireEmitter));
+        context.subscriptions.push(vscode.debug.onDidTerminateDebugSession(fireEmitter));
+
+        context.subscriptions.push(this.rascalDebugClient.portRegistrationEvent(fireEmitter));
+
+        this.context.subscriptions.push(
+            vscode.commands.registerCommand("rascalmpl.startDebuggerForRepl", (replNode: RascalReplNode) => {
+                if (replNode.serverPort !== undefined) {
+                    this.rascalDebugClient.startDebuggingSession(replNode.serverPort);
+                }
+            })
+        );
+    }
+
+    getTreeItem(element: RascalReplNode): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        return element;
+    }
+    getChildren(element?: RascalReplNode | undefined): vscode.ProviderResult<RascalReplNode[]> {
+        if (element === undefined) {
+            return this.updateRascalDebugView();
+        }
+        return [];
+    }
+    getParent?(_element: RascalReplNode): vscode.ProviderResult<RascalReplNode> {
+        //The Rascal debug view gives a flat list of the opened Rascal terminals. As such, only root items exit in this view.
+        return undefined;
+    }
+    resolveTreeItem?(item: vscode.TreeItem, _element: RascalReplNode, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.TreeItem> {
+        return item;
+    }
+
+    async makeLabel(label: string, processId : number) : Promise<string | vscode.TreeItemLabel> {
+        if (await this.isActiveTerminal(processId)) {
+            return {label: label, highlights : [[0, label.length]]};
+        }
+        return label;
+    }
+
+    async updateRascalDebugView() : Promise<RascalReplNode[]> {
+        const result : RascalReplNode[] = [];
+        for (const terminal of vscode.window.terminals) {
+            const processId = await terminal.processId;
+            if (processId === undefined) {
+                continue;
+            }
+            if (terminal.name.includes("Rascal terminal")) {
+                const label = await this.makeLabel(terminal.name, processId);
+                const serverPort = this.rascalDebugClient.getServerPort(processId);
+                const isDebugging = serverPort !== undefined && this.rascalDebugClient.isConnectedToDebugServer(serverPort);
+                const replNode = new RascalReplNode(label, serverPort, isDebugging);
+                if (serverPort !== undefined && !isDebugging) {
+                    replNode.contextValue = "canStartDebugging";
+                }
+                result.push(replNode);
+            }
+        }
+        return result;
+    }
+
+    async isActiveTerminal(processId : number) : Promise<boolean> {
```

Comment:
since we're rebuiling this list quite frequently (instead of only precisly invalidating the entry we know has changed), I would like to propose we don't `await` on `activeTerminal.processId` for every entry in the loop of open terminals, but determine this once (inside `updateRascalDebugView` function), and pass a boolean into the `makeLabel` function.